.github/workflows/reusable_pre_release.yaml

```diff
@@ -0,0 +1,129 @@
+        required: true
+        type: string
+      test_package:
+        description: Defines if the package should be tested using the 'newrelic/integrations-pkg-test-action' action.
+        type: boolean
+        required: false
+        default: true
+      publish_schema:
+        description: Defines the schema for the publish_action.
+        type: string
+        required: false
+        default: "ohi"
+      publish_schema_url:
+        description: Defines the schema_url for the publish_action (it requires publish_schema='custom')
+        type: string
+        required: false
+        default: ""
+      run_test_windows:
+        required: false
+        type: boolean
+        default: true
+      run_integration_nix:
+        required: false
+        type: boolean
+        default: true
+      run_build-win-packages:
+        description: "Build Windows Packages"
+        required: false
+        type: boolean
+        default: true
+      win_package_type:
+        description: "Package type (msi,exe). Integrations that wraps other binaries are packages in exe (nri-kafka)"
+        default: msi
+        required: false
+        type: string
+    secrets:
+      OHAI_AWS_ROLE_ARN_STAGING:
+        required: true
+      OHAI_AWS_ROLE_SESSION_NAME_STAGING:
+        required: true
+      OHAI_AWS_ACCESS_KEY_ID_STAGING:
+        required: true
+      OHAI_AWS_SECRET_ACCESS_KEY_STAGING:
+        required: true
+      OHAI_GPG_PASSPHRASE:
+        required: true
+      OHAI_GPG_PRIVATE_KEY_BASE64:
+        required: true
+      OHAI_PFX_CERTIFICATE_BASE64:
+        required: true
+      OHAI_PFX_PASSPHRASE:
+        required: true
+      COREINT_SLACK_TOKEN:
+        required: true
+      COREINT_SLACK_CHANNEL:
+        required: true
+
+jobs:
+  # TODO add the same conditions of pre-release on the security workflow in the repo.
+
+  test-nix:
+    uses: ./.github/workflows/_test_nix.yaml
+
+  test-windows:
+    if: ${{ inputs.run_test_windows }}
+    uses: ./.github/workflows/_test_windows.yaml
+
+  test-integration-nix:
+    if: ${{ inputs.run_integration_nix }}
+    uses: ./.github/workflows/_test_integration_nix.yaml
+
+  builds-packages:
+    needs: [test-integration-nix,test-windows,test-nix]
+    uses: ./.github/workflows/_pre_release_build.yaml
+    permissions:
+      # needed to upload release artifacts.
+      contents: write
+    with:
+      tag: ${{ inputs.tag }}
+      integration: ${{ inputs.integration }}
+      test_package: ${{ inputs.test_package }}
+      run_build-win-packages: ${{ inputs.run_build-win-packages }}
+      win_package_type: ${{ inputs.win_package_type }}
+    secrets:
+      OHAI_GPG_PASSPHRASE: ${{ secrets.OHAI_GPG_PASSPHRASE }}
+      OHAI_GPG_PRIVATE_KEY_BASE64: ${{ secrets.OHAI_GPG_PRIVATE_KEY_BASE64 }}
+      OHAI_PFX_CERTIFICATE_BASE64: ${{ secrets.OHAI_PFX_CERTIFICATE_BASE64 }}
+      OHAI_PFX_PASSPHRASE: ${{ secrets.OHAI_PFX_PASSPHRASE }}
+  
+  # publish:
```

Comment:
Since in the release I kept it directly in the reusable file, does it makes sense to extract it in this case?
If so maybe we should extract it also in the other one. 

However, I fear we are adding too many layers for something is difficult to test